src/test/scala/com/datamountaineer/streamreactor/connect/schemas/TestConverterUtil.scala

```diff
@@ -233,5 +233,262 @@ class TestConverterUtil extends TestUtilsBase with ConverterUtil {
+        fields = Map.empty,
+        ignoreFields = Set.empty[String],
+        keyFields = Map.empty,
+        headerFields = Map("*" -> "*")
+      )
+
+      val expected = "{\"header_field_1\":\"foo\",\"header_field_2\":\"bar\",\"header_field_3\":\"boo\"}"
+
+      combinedRecord.valueSchema().fields().asScala.size shouldBe originalRecord.headers().size
+
+      simpleJsonConverter.fromConnectData(combinedRecord.valueSchema(), combinedRecord.value()).toString shouldBe expected
+    }
+
+    "should return empty record for struct" in {
+
+      val originalRecord = sinkRecordWithKeyHeaders()
+      val combinedRecord = originalRecord.newFilteredRecord(
+        fields = Map.empty,
+        ignoreFields = Set.empty[String],
+        keyFields = Map.empty,
+        headerFields = Map.empty
+      )
+
+      combinedRecord.valueSchema().fields().asScala.size shouldBe 0
+    }
+
+    "should retain key and headers for struct" in {
+
+      val originalRecord = sinkRecordWithKeyHeaders()
+      val combinedRecord = originalRecord.newFilteredRecord(
+        fields = Map.empty,
+        ignoreFields = Set.empty[String],
+        keyFields = Map.empty,
+        headerFields = Map.empty,
+        retainKey = true,
+        retainHeaders = true
+      )
+
+      combinedRecord.valueSchema().fields().asScala.size shouldBe 0
+      combinedRecord.keySchema() shouldBe originalRecord.keySchema()
+      combinedRecord.key() shouldBe originalRecord.key()
+      combinedRecord.headers() shouldBe originalRecord.headers()
+    }
+
+    "should ignore fields for struct" in {
+      val originalRecord = getTestRecord
+      val combinedRecord = originalRecord.newFilteredRecord(
+        fields = Map("*" -> "*"),
+        ignoreFields = Set("int_field"),
+        keyFields = Map.empty,
+        headerFields = Map.empty,
+        retainKey = true,
+        retainHeaders = true
+      )
+
+      val expected = "{\"id\":\"sink_test-1-1\",\"long_field\":1,\"string_field\":\"foo\"}"
+
+      simpleJsonConverter.fromConnectData(combinedRecord.valueSchema(), combinedRecord.value()).toString shouldBe expected
+    }
+
+    "should add fields, key fields and headers for schemaless JSON" in {
+      val fields = new util.HashMap[String, Any]()
+      fields.put("field1", "value1")
+      fields.put("field2", 3)
+      fields.put("field3", null)
+
+      val keys = new util.HashMap[String, Any]()
+      keys.put("key_field1", "key_value1")
+      keys.put("key_field2", 3)
+      keys.put("key_field3", null)
+      val record = new SinkRecord("t", 0, null, keys, null, fields, 0)
+
+      val combinedRecord = record.newFilteredRecord(
+        fields = Map("field2" -> "field2_alias"),
+        ignoreFields = Set.empty,
+        keyFields = Map("key_field1" -> "key_field1_alias"),
+        headerFields = Map.empty,
+        retainKey = true,
+        retainHeaders = true
+      )
+
+      val expected = "{\"key_field1_alias\":\"key_value1\",\"field2_alias\":\"3\"}"
```

Comment:
this type change where we default to STRING might be a problem - imagine we move a field which is number to string to MongoDB and then the user queries on the database won't work since they  expect a number